packages/guides/tests/unit/Renderer/UrlGenerator/RelativeUrlGeneratorTest.php

```diff
@@ -0,0 +1,115 @@
+            ],
+            'Same File with anchor' => [
+                'expected' => '#anchor',
+                'canonicalUrl' => 'directory/file.html#anchor',
+                'currentPath' => 'directory/file',
+            ],
+            'File in same directory' => [
+                'expected' => 'file.html',
+                'canonicalUrl' => 'directory/file.html',
+                'currentPath' => 'directory/anotherFile',
+            ],
+            'File in subdirectory' => [
+                'expected' => 'subdirectory/file.html',
+                'canonicalUrl' => 'directory/subdirectory/file.html',
+                'currentPath' => 'directory/anotherFile',
+            ],
+            'File in directory above' => [
+                'expected' => '../file.html',
+                'canonicalUrl' => 'file.html',
+                'currentPath' => 'directory/anotherFile',
+            ],
+            'File in two directory above' => [
+                'expected' => '../../file.html',
+                'canonicalUrl' => 'file.html',
+                'currentPath' => 'directory/subdirectory/anotherFile',
+            ],
+            'File in other subdirectory above' => [
+                'expected' => '../subdirectory/file.html',
+                'canonicalUrl' => 'directory/subdirectory/file.html',
+                'currentPath' => 'directory/othersubdirectory/anotherFile',
+            ],
+        ];
+    }
+
+    #[DataProvider('fileUrlProvider')]
+    public function testCreateFileUrl(string $expected, string $filename, string $outputFormat = 'html', string|null $anchor = null, string $skip = ''): void
+    {
+        if ($skip !== '') {
+            self::markTestSkipped($skip);
+        }
+
+        $urlGenerator = new RelativeUrlGenerator();
+        self::assertSame($expected, $urlGenerator->createFileUrl($filename, $outputFormat, $anchor));
+    }
+
+    /** @return array<string, array<string, string|null>> */
+    public static function fileUrlProvider(): array
+    {
+        return [
+            'Simple Filename' => [
+                'expected' => 'file.html',
+                'filename' => 'file',
+            ],
+            'Complex Filename' => [
+                'expected' => 'file-something.html',
+                'filename' => 'file-something',
+            ],
+            'Output Format' => [
+                'expected' => 'texfile.tex',
+                'filename' => 'texfile',
+                'outputFormat' => 'tex',
+            ],
+            'File with anchor' => [
+                'expected' => 'file.html#anchor',
+                'filename' => 'file',
+                'outputFormat' => 'html',
+                'anchor' => 'anchor',
+            ],
+            'Empty File with anchor' => [
+                'expected' => '#anchor',
+                'filename' => '',
+                'outputFormat' => 'html',
+                'anchor' => 'anchor',
+                'skip' => 'Empty filenames are not supported',
```

Comment:
can we support this in the future? or is this an error situation?

Reply:
Yes sure. Just vheck if the document name is empty?